benchmark/domain/synthetic_3.py

```diff
@@ -0,0 +1,94 @@
+"""Synthetic dataset. Custom parabolic test with irrelevant parameters."""
+
+from uuid import UUID
+
+from numpy import pi, sin, sqrt
+from pandas import DataFrame
+
+from baybe.campaign import Campaign
+from baybe.objective import SingleTargetObjective
+from baybe.parameters import NumericalContinuousParameter, NumericalDiscreteParameter
+from baybe.recommenders import RandomRecommender
+from baybe.searchspace import SearchSpace
+from baybe.simulation import simulate_scenarios
+from baybe.targets import NumericalTarget, TargetMode
+from benchmark.definition import Benchmark
+
+
+def lookup_synthetic_3(z: int, x: float, y: float) -> float:
+    """Lookup function for the synthetic dataset."""
+    assert -2 * pi <= x <= 2 * pi
+    assert -2 * pi <= y <= 2 * pi
+    assert z in {1, 2, 3, 4}
+
+    if z == 1:
+        return sin(x) * (1 + sin(y))
+    if z == 2:
+        return x * sin(0.9 * x) + sin(x) * sin(y)
+    if z == 3:
+        return sqrt(x + 8) * sin(x) + sin(x) * sin(y)
+    if z == 4:
+        return x * sin(1.666 * sqrt(x + 8)) + sin(x) * sin(y)
+    raise ValueError("Invalid z value.")
+
+
+def synthetic_3() -> tuple[DataFrame, dict[str, str]]:
+    """Synthetic dataset.
+
+        Number of Samples            inf
+        Dimensionality                 3
+        Features:
+            z   discrete {1,2,3,4}
+            x   continuous [-2*pi, 2*pi]
+            y   continuous [-2*pi, 2*pi]
+        Targets:
+            output   continuous
+    Best Value 4.09685
+    """
+    synthetic_3_continues = [
+        NumericalContinuousParameter("x", (-2 * pi, 2 * pi)),
+        NumericalContinuousParameter("y", (-2 * pi, 2 * pi)),
+        NumericalDiscreteParameter("z", (1, 2, 3, 4)),
+    ]
+    objective = SingleTargetObjective(
+        target=NumericalTarget(name="output", mode=TargetMode.MAX)
+    )
+
+    campaign = Campaign(
+        searchspace=SearchSpace.from_product(parameters=synthetic_3_continues),
+        objective=objective,
+    )
+    campaign_rand = Campaign(
+        searchspace=SearchSpace.from_product(parameters=synthetic_3_continues),
+        recommender=RandomRecommender(),
+    )
+
+    batch_size = 5
+    n_doe_iterations = 30
+    n_mc_iterations = 50
+
+    metadata = {
```

Comment:
is there anyr eason we want to have metadata as `dict`? If we need flexibiloty I think thats fine, but if we just always have these three or a fixed set of always the same variables why not make a little `Metadata` class

Reply:
The intention from my end was flexibility. But I see your point also for the commnant above. Maybe we should discuss some kind of domain object type to encapulate parameters, goals and so and a config object for repeated settings for the benchmark for one of the next PRs. What do you think?

Reply:
I think we should have a general discussion on the intended workflow for these benchmarks in the dev meeting and that will probably shed some light on what is a good structure 👍🏼